featuregates/client/go.mod

```diff
@@ -11,6 +11,7 @@ replace (
 	github.com/vmware-tanzu/tanzu-framework/cli/core => ../../cli/core
 	github.com/vmware-tanzu/tanzu-framework/cli/runtime => ../../cli/runtime
 	github.com/vmware-tanzu/tanzu-framework/packageclients => ../../packageclients
+	github.com/vmware-tanzu/tanzu-framework/pkg/v1/tkr => ../../pkg/v1/tkr
```

Comment:
don't we need to avoid this dependency?

Reply:
i refactored this go.mod to not include this module, i will push the changes in this pr https://github.com/vmware-tanzu/tanzu-framework/pull/3613, i created this pr just to get review on the API design.. actual API changes would go here https://github.com/vmware-tanzu/tanzu-framework/pull/3613